starlette/middleware/__init__.py

```diff
@@ -1,12 +1,28 @@
-import typing
+import sys
+from typing import Any, Callable, Iterator
+
+from starlette.types import ASGIApp
+
+if sys.version_info >= (3, 10):  # pragma: no cover
+    from typing import Concatenate, ParamSpec
+else:  # pragma: no cover
+    from typing_extensions import Concatenate, ParamSpec
+
+
+P = ParamSpec("P")
 
 
 class Middleware:
-    def __init__(self, cls: type, **options: typing.Any) -> None:
+    def __init__(
+        self,
+        cls: Callable[Concatenate[ASGIApp, P], Any],
+        *args: P.args,
```

Comment:
We either need to make it work with `args` or raise a `RuntimeError`.

Reply:
I vote make it work. As per above I'm _guessing_ the main reason to not allow args is that it's more error prone than kwargs but that's very much mitigated by enforcing typing. And args can be more convenient in simple cases!

Reply:
I agree. I've just pushed changes regarding making it work with `*args`. Please have another look.

One thought comes to mind - enabling the usage of `*args` is somewhat a breaking change if one is directly using `Middleware`'s `__iter__` or `__repr__`. What's your take on this? Do you consider this a breaking change?

Reply:
It's ok.